crates/issue/src/lib.rs

```diff
@@ -301,13 +305,75 @@ impl<T: Config> Pallet<T> {
         T::TreasuryPalletId::get().into_account_truncating()
     }
 
+    // Function to cancel an issue request and slash collateral
+    pub fn cancel_issue_request_and_slash_collateral(issue_id: &H256) -> Result<BalanceOf<T>, DispatchError> {
```

Comment:
@gregdhill should we have griefing collateral at all? We already have the redeem punishment so it's not adding a lot (but if it's simplifying code, maybe it's ok)

Reply:
can you elaborate on what exactly you mean by redeem punishment as the `_request_redeem` for replace will not lock `old_vault` collateral, it will only lock if it is a normal `request_redeem` call, [here](https://github.com/interlay/interbtc/blob/4a0f29b4162721bb01299722de864855de7a8b8e/crates/redeem/src/lib.rs#L635) is the condition.

Reply:
The `cancel_issue_request_and_slash_collateral` function has been removed as it is unused and `cancel_issue`  will handle the logic.

Reply:
> @gregdhill should we have griefing collateral at all? We already have the redeem punishment so it's not adding a lot (but if it's simplifying code, maybe it's ok)



> When the Vault fails to complete a redeem request it is slashed, this should be the same case here; if the `old_vault` does not relay the BTC tx before expiry the `new_vault` should be compensated. Therefore we probably don't need griefing collateral for `request_replace` since then on failure we can slash the already locked collateral on cancel.


For griefing collateral following is the flow

In the current solution, when `old_vault` calls `request_replace` extrinsic 
- `_request_issue` will lock `old_vault` griefing collateral 
- `_request_redeem` doesn't lock any griefing collateral for replace calls

If the `request_replace` is executed correctly by calling `execute_redeem` and submitting BTC proof
- `_complete_vault_issue` will unlock collateral 

If the `request_replace` is not executed correctly and `cancel_redeem` is called 
- `_cancel_issue` will transfer lock griefing collateral from `old_vault` to `new_vault`

Reply:
This should be solved in issue #1182

Reply:
Based on the discussion in the call it is better if we resolve it in this PR itself.